src/Storage/Bucket.php

```diff
@@ -671,4 +739,28 @@ public function name()
     {
         return $this->identity['bucket'];
     }
+
+    /**
+     * Returns whether the bucket with the given file prefix is writable.
+     * Tries to create a temporary file as a resumable upload which will
+     * not be completed (and cleaned up by GCS).
+     *
+     * @param  string $file Optional file to try to write.
+     * @return boolean
+     */
+    public function isWritable($file = null)
```

Comment:
403's are currently `ServiceException`s, so we'll just rethrow the exception if it's not a 403.